sdks/java/maven-archetypes/examples/src/main/resources/archetype-resources/src/main/java/common/ExampleUtils.java

```diff
@@ -91,19 +79,26 @@ public DataflowExampleUtils(DataflowPipelineOptions options, boolean isUnbounded
    * @throws IOException if there is a problem setting up the resources
    */
   public void setup() throws IOException {
-    setupPubsubTopic();
-    setupBigQueryTable();
-  }
-
-  /**
-   * Set up external resources, and configure the runner appropriately.
-   */
-  public void setupResourcesAndRunner(boolean isUnbounded) throws IOException {
-    if (isUnbounded) {
-      options.setStreaming(true);
+    Sleeper sleeper = Sleeper.DEFAULT;
+    BackOff backOff =
+        FluentBackoff.DEFAULT
+            .withMaxRetries(3).withInitialBackoff(Duration.millis(200)).backoff();
+    Throwable lastException = null;
+    try {
+      do {
+        try {
+          setupPubsub();
+          setupBigQueryTable();
+          return;
+        } catch (GoogleJsonResponseException e) {
+          lastException = e;
+        }
+      } while (BackOffUtils.next(sleeper, backOff));
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      // Ignore InterruptedException
```

Comment:
What happens if an `InterruptedException` is thrown before any other exception? The loop will terminate, so we'll throw a `RuntimeException`, so we should probably handle that case explicitly here.